### DISCUSSION TITLE
cumulative time in time_callback

### DISCUSSION THREAD

**Discussion by hovjr** [google/or-tools#3210] [Routing (and legacy CP) questions] [answered]
I am in a position in which I need a cumulative time to be used each time the run reaches the next node (for the purpose of retrieving an updated time from an external file due to adjust for rush hour for example).

ie, i run a tsp for the first time during which i record the current time and what i need is for the next node/time_callback function, is to have the cumulative time along the tsp path. 

hope this is clear,
like below:

def time_callback(from_index, to_index):
    *cumulative_time*
    from_node = manager.IndexToNode(from_index)
    to_node = manager.IndexToNode(to_index)
    time_delay = np.select(external[timeframe] == cumulative_time, external[time_delay])
    output = df['time_matrix'][from_node][to_node] + time_delay
    return output

**Comment by Mizux** [comment, marked as answer]
this is call the dimension dependent dimension, it's only available in C++ and it's extremely slow (basically solver will build the N^2 transit matrix M time, with M being the range of value from 0 to vehicle capacity)....

**Reply by hovjr** [reply]
Is there a way to achieve what I am describing? Thank you for your input.

**Reply by Mizux** [reply]
you can found an "outdated" (v6.9 ?) example here: https://github.com/google/or-tools/issues/574#issuecomment-430203076

**Reply by lperron** [reply]
I would not use it.
